frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/overview/workspace_overview_block_component.dart

```diff
@@ -0,0 +1,415 @@
+import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/icon.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
+import 'package:appflowy_editor/appflowy_editor.dart' hide Log;
+import 'package:collection/collection.dart';
+import 'package:dartz/dartz.dart' hide State;
+import 'package:easy_localization/easy_localization.dart' hide TextDirection;
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flowy_infra_ui/style_widget/hover.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class WorkspaceOverviewBlockKeys {
+  const WorkspaceOverviewBlockKeys._();
+
+  static const String type = 'overview';
+  static const String backgroundColor = blockComponentBackgroundColor;
+}
+
+SelectionMenuItem workspaceOverviewItem = SelectionMenuItem.node(
+  getName: () => LocaleKeys.document_selectionMenu_workspaceOverview.tr(),
+  iconData: Icons.list_alt,
+  keywords: ['overview', 'workspace overview'],
+  nodeBuilder: (editorState, _) => overviewBlockNode(),
+  replace: (_, node) => node.delta?.isEmpty ?? false,
+);
+
+Node overviewBlockNode() => Node(type: WorkspaceOverviewBlockKeys.type);
+
+class WorkspaceOverviewBlockComponentBuilder extends BlockComponentBuilder {
+  WorkspaceOverviewBlockComponentBuilder({
+    super.configuration,
+  });
+
+  @override
+  BlockComponentWidget build(BlockComponentContext blockComponentContext) {
+    final node = blockComponentContext.node;
+    return WorkspaceOverviewBlockWidget(
+      key: node.key,
+      node: node,
+      configuration: configuration,
+      showActions: showActions(node),
+      actionBuilder: (_, state) => actionBuilder(blockComponentContext, state),
+    );
+  }
+
+  @override
+  bool validate(Node node) => node.children.isEmpty;
+}
+
+class WorkspaceOverviewBlockWidget extends BlockComponentStatefulWidget {
+  const WorkspaceOverviewBlockWidget({
+    super.key,
+    required super.node,
+    super.showActions,
+    super.actionBuilder,
+    super.configuration = const BlockComponentConfiguration(),
+  });
+
+  @override
+  State<BlockComponentStatefulWidget> createState() =>
+      _OverviewBlockWidgetState();
+}
+
+class _OverviewBlockWidgetState extends State<WorkspaceOverviewBlockWidget>
+    with
+        BlockComponentConfigurable,
+        BlockComponentTextDirectionMixin,
+        BlockComponentBackgroundColorMixin,
+        TickerProviderStateMixin {
+  @override
+  BlockComponentConfiguration get configuration => widget.configuration;
+
+  @override
+  EditorState get editorState => context.read<EditorState>();
+
+  @override
+  Node get node => widget.node;
+
+  static const double startingLeftIndent = 50.0;
+
+  static const double leftIndentIncrementValue = 14.0;
+
+  String get viewId => context.read<DocumentBloc>().view.id;
+
+  final ValueNotifier<bool> _isExpandedNotifier = ValueNotifier(true);
```

Comment:
imo, the "expand" value should also be saved in the data because this value needs to be synchronized.

Reply:
**@LucasXu0**, Could you provide additional context regarding the specific 'data' you're referring to?